src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.subsystems.intake;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+import edu.wpi.first.wpilibj.PneumaticsModuleType;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+
+public class Intake extends SubsystemBase {
+    private static Intake INSTANCE;
+    private final WPI_TalonSRX motor = new WPI_TalonSRX(Ports.Intake.MOTOR);
+    private final Solenoid solenoid = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Intake.SOLENOID);
+
+    private Intake() {
+        motor.setInverted(Ports.Intake.IS_INVERTED);
+    }
+
+
+    /**
+     * Lazy instantiation.
+     *
+     * @return the subsystem instance.
+     */
+    public static Intake getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Intake();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * Sets the power of the intake [%].
+     *
+     * @param power desired power in percentage.
+     */
+    public void setPower(double power) {
+        motor.set(ControlMode.PercentOutput, power);

```

Comment:
This makes things more complicated, but would closed-loop control help here? @adamv17

Reply:
I think pid for the intake is most likely unnecessary, the idea in the code is that based on the speed the robot is traveling the intake will receive a different power (we don't know if it is less or more)

Reply:
I don't think velocity control is really necessary for the intake, and probably percent output would work just fine.